backend/dataall/core/environment/services/environment_service.py

```diff
@@ -120,120 +150,189 @@ def validate_permissions(session, uri, g_permissions, group):
+            log.info('Check if PivotRole exist in the account')
+            pivot_role_arn = SessionHelper.get_delegation_role_arn(accountid=account_id, region=region)
+            role = IAM.get_role(
+                account_id=account_id, region=region, role_arn=pivot_role_arn, role=cdk_look_up_role_arn
+            )
+            if not role:
+                raise exceptions.AWSResourceNotFound(
+                    action='CHECK_PIVOT_ROLE',
+                    message='Pivot Role has not been created in the Environment AWS Account',
+                )
+
+        mlStudioEnabled = None
+        for parameter in data.get('parameters', []):
+            if parameter['key'] == 'mlStudiosEnabled':
+                mlStudioEnabled = parameter['value']
+
+        if mlStudioEnabled and data.get('vpcId', None) and data.get('subnetIds', []):
+            log.info('Check if ML Studio VPC Exists in the Account')
+            EC2.check_vpc_exists(
+                AwsAccountId=account_id,
+                region=region,
+                role=cdk_look_up_role_arn,
+                vpc_id=data.get('vpcId', None),
+                subnet_ids=data.get('subnetIds', []),
+            )
+
+        return cdk_role_name
+
     @staticmethod
     @TenantPolicyService.has_tenant_permission(MANAGE_ENVIRONMENTS)
     @ResourcePolicyService.has_resource_permission(LINK_ENVIRONMENT)
-    def create_environment(session, uri, data=None):
+    def create_environment(uri, data=None):
         context = get_context()
-        EnvironmentRequestValidationService.validate_creation_params(data, uri, session)
-        organization = OrganizationRepository.get_organization_by_uri(session, uri)
-        env = Environment(
-            organizationUri=data.get('organizationUri'),
-            label=data.get('label', 'Unnamed'),
-            tags=data.get('tags', []),
-            owner=context.username,
-            description=data.get('description', ''),
-            environmentType=data.get('type', EnvironmentType.Data.value),
-            AwsAccountId=data.get('AwsAccountId'),
-            region=data.get('region'),
-            SamlGroupName=data['SamlGroupName'],
-            validated=False,
-            isOrganizationDefaultEnvironment=False,
-            EnvironmentDefaultIAMRoleName=data.get('EnvironmentDefaultIAMRoleArn', 'unknown').split('/')[-1],
-            EnvironmentDefaultIAMRoleArn=data.get('EnvironmentDefaultIAMRoleArn', 'unknown'),
-            CDKRoleArn=f"arn:aws:iam::{data.get('AwsAccountId')}:role/{data['cdk_role_name']}",
-            resourcePrefix=data.get('resourcePrefix'),
-        )
+        with context.db_engine.scoped_session() as session:
+            EnvironmentRequestValidationService.validate_creation_params(data, uri, session)
+            cdk_role_name = EnvironmentService.check_cdk_resources(data.get('AwsAccountId'), data.get('region'), data)
+            organization = OrganizationRepository.get_organization_by_uri(session, uri)
```

Comment:
nit: is organization used? Can we remove like call `get_organization_by_uri()` ?

Reply:
It was originally used for logging purposes, so I decided to keep it. 
Now in logs I used uri instead of name.